### PR TITLE
Added sortStringArray utility function

### PR DESCRIPTION
**Pull request by codyebberson**
Prep work for https://github.com/medplum/medplum/pull/4296

Failed on sonar warning:

image

### CODE REVIEW

File: packages/core/src/utils.ts [hunk 1]
```diff
@@ -1056,3 +1056,12 @@ export function append<T>(array: T[] | undefined, value: T): T[] {
 export function getWebSocketUrl(path: string, baseUrl: URL | string): string {
   return new URL(path, baseUrl).toString().replace('http://', 'ws://').replace('https://', 'wss://');
 }
+
+/**
+ * Sorts an array of strings using the localeCompare method.
```
Comment (ThatOneBro) [inline packages/core/src/utils.ts:1061]: Can we also note that it mutates and returns the input array?

File: packages/server/src/fhir/patient.ts [hunk 2]
```diff
@@ -72,9 +72,7 @@ export function getPatients(resource: Resource): Reference<Patient>[] {
       }
     }
   }
-  return Array.from(result)
-    .sort((a, b) => a.localeCompare(b))
-    .map((reference) => ({ reference }));
+  return Array.from(result).map((reference) => ({ reference }));
```
Comment (ThatOneBro) [inline packages/server/src/fhir/patient.ts:75]: Is it intentional that the sort was removed? 👀

Comment (codyebberson) [reply packages/server/src/fhir/patient.ts:75]: There's no logical reason for this method to return sorted results.  It's sorting on `reference`, which is effectively sorting on v4 UUIDs, which is kinda arbitrary.

It does look like the tests assumed sorted results.... I will look into whether we need to update those.

File: packages/core/src/utils.ts (hunk 1 above)
Comment (codyebberson) [reply packages/core/src/utils.ts]: ✅  comments updated

File: packages/server/src/fhir/patient.ts (hunk 2 above)
Comment (codyebberson) [reply packages/server/src/fhir/patient.ts:75]: ✅  tests updated

**Review by ThatOneBro — APPROVED** [review]

**Pull request merged**